strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/indexing/local/ArtifactEntryMinimalArtifactInfoIndexCreator.java

```diff
@@ -0,0 +1,129 @@
+package org.carlspring.strongbox.storage.indexing.local;
+
+import org.carlspring.strongbox.artifact.coordinates.MavenArtifactCoordinates;
+import org.carlspring.strongbox.domain.ArtifactEntry;
+
+import java.util.Map;
+
+import org.apache.commons.collections4.MapUtils;
+import org.apache.maven.index.ArtifactAvailability;
+import org.apache.maven.index.ArtifactContext;
+import org.apache.maven.index.ArtifactInfo;
+import org.apache.maven.index.creator.MinimalArtifactInfoIndexCreator;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+@Order(1)
+public class ArtifactEntryMinimalArtifactInfoIndexCreator
+        extends MinimalArtifactInfoIndexCreator
+{
+
+    public static final ArtifactEntryMinimalArtifactInfoIndexCreator INSTANCE = new ArtifactEntryMinimalArtifactInfoIndexCreator();
+
+    private ArtifactEntryMinimalArtifactInfoIndexCreator()
+    {
+        super();
+    }
+
+    public void populateArtifactInfo(ArtifactContext artifactContext)
+    {
+        ArtifactEntryArtifactContext ac = (ArtifactEntryArtifactContext) artifactContext;
+        ArtifactEntry artifactEntry = ac.getArtifactEntry();
+        ArtifactInfo artifactInfo = ac.getArtifactInfo();
+
+        MavenArtifactCoordinates coordinates = (MavenArtifactCoordinates) artifactEntry.getArtifactCoordinates();
+
+        ArtifactInfo ai = ac.getArtifactInfo();
+
+        // TODO handle artifacts without poms
+        if (ac.pomExists())
+        {
+            if (ai.getClassifier() != null)
+            {
+                ai.setSourcesExists(ArtifactAvailability.NOT_AVAILABLE);
+
+                ai.setJavadocExists(ArtifactAvailability.NOT_AVAILABLE);
+            }
+            else
+            {
+                boolean sourcesExists = ac.sourcesExists();
+                if (!sourcesExists)
+                {
+                    ai.setSourcesExists(ArtifactAvailability.NOT_PRESENT);
+                }
+                else
+                {
+                    ai.setSourcesExists(ArtifactAvailability.PRESENT);
+                }
+
+                boolean javadocExists = ac.javadocExists();
+                if (!javadocExists)
+                {
+                    ai.setJavadocExists(ArtifactAvailability.NOT_PRESENT);
+                }
+                else
+                {
+                    ai.setJavadocExists(ArtifactAvailability.PRESENT);
+                }
+            }
+        }
+
+        /* We do not support this. Should we ?
+        Model model = ac.getPomModel();
```

Comment:
Clarification: Is this section copied from the Maven Indexer?

Reply:
yes, maven indexer additionally indexes `model.name` and `model.description`, but we don't store this information in `ArtifactEntry`. I think it'd be not used, `GAV` is the essence, right ?

Reply:
Yeah, you're right.